### PR TITLE
OCPBUGS-22487: Fix operands list endpoint.

### PR DESCRIPTION
**Pull request by TheRealJon**
The operands list endpoint was proxying to the public API server URL instead of the proxy endpoint configured at startup. This was causing mismatched TLS configuration when the console is running on-cluster.

### CODE REVIEW

File: pkg/server/server.go [hunk 1]
```diff
@@ -460,7 +461,7 @@ func (s *Server) HTTPHandler() (http.Handler, error) {
 
 	// List operator operands endpoint
 	operandsListHandler := &OperandsListHandler{
-		APIServerURL: s.KubeAPIServerURL,
+		APIServerURL: k8sProxyURL,
```
Comment (jhadvig) [inline pkg/server/server.go:464]: Screenshot 2024-02-27 at 14 21 00

Doing some detective work, I see this is my fault. Nice catch @TheRealJon 👍

File: pkg/server/server.go [hunk 2]
```diff
@@ -166,7 +166,7 @@ type Server struct {
 	K8sProxyConfig                      *proxy.Config
 	KnativeChannelCRDLister             ResourceLister
 	KnativeEventSourceCRDLister         ResourceLister
-	KubeAPIServerURL                    string
+	KubeAPIServerURL                    string // JS global only. Not used for proxying.
```
Comment (jhadvig) [inline pkg/server/server.go:169]: I think we should have a followup story which would extract the `Server` type into the `pkg/serverconfig/types.go` and add proper go docs to each of these fields to clarify their usage, maybe even with an example. I know that we are not un/marshalling the type anywhere in the code but I feel that we came to a point where we need to have docs for our types cause its hard to keep track with all details.
WDYT ?

**Review by jhadvig — APPROVED** [review]
/lgtm
/approve

File: pkg/server/server.go (hunk 2 above)
Comment (TheRealJon) [reply pkg/server/server.go:169]: Yeah, makes sense to me.

**Pull request merged**